src/playercards/CardsThatSealTokens.ttslua

```diff
@@ -0,0 +1,154 @@
+    self.addContextMenuItem("Release token", releaseOneToken)
+  elseif SHOW_MULTI_RELEASE then
+    self.addContextMenuItem("Release " .. SHOW_MULTI_RELEASE .. " token(s)", function(playerColor)
+      if SHOW_MULTI_RELEASE >= #sealedTokens then
+        for i = 1, SHOW_MULTI_RELEASE do
+          releaseOneToken(playerColor)
+        end
+      else
+        printToColor("Not enough " .. name .. " tokens sealed.", playerColor)
+      end
+    end)
+  else
+    self.addContextMenuItem("Release token(s)", releaseTokens)
+  end
+
+  -- main context menu options to seal tokens
+  for _, map in pairs(ID_URL_MAP) do
+    if (VALID_TOKENS[map.name] ~= nil) or (SHOW_READ_BAG and tokensInBag[map.name] and not INVALID_TOKENS[map.name]) then
+      if not SHOW_MULTI_SEAL then
+        self.addContextMenuItem("Seal " .. map.name, function(playerColor)
+          sealToken(map.name, playerColor)
+        end)
+      else
+        self.addContextMenuItem("Seal " .. SHOW_MULTI_SEAL .. " " .. map.name, function(playerColor)
+          readBag()
+          local allowed = true
+          local notFound
+
+          for name, _ in pairs(VALID_TOKENS) do
+            if (tokensInBag[name] or 0) < SHOW_MULTI_SEAL then
+              allowed = false
+              notFound = name
+            end
+          end
+
+          if allowed then
+            for i = 1, SHOW_MULTI_SEAL do
+              sealToken(map.name, playerColor)
+            end
+          else
+            printToColor("Not enough " .. notFound .. " tokens in the chaos bag.", playerColor)
+          end
+        end)
+      end
+    end
+  end
+end
+
+-- generates a list of chaos tokens that is in the chaos bag
+function readBag()
+  local chaosbag = Global.call("findChaosBag")
+  tokensInBag = {}
+
+  for _, token in ipairs(chaosbag.getObjects()) do
+    tokensInBag[token.name] = (tokensInBag[token.name] or 0) + 1
+  end
+end
+
+-- seals the named token on this card
+function sealToken(name, playerColor)
+  if not Global.call("canTouchChaosTokens") then return end
+  local chaosbag = Global.call("findChaosBag")
+  for i, obj in ipairs(chaosbag.getObjects()) do
+    if obj.name == name then
+      chaosbag.takeObject({
+        position = self.getPosition() + Vector(0, 0.5 + 0.1 * #sealedTokens, 0),
+        rotation = self.getRotation(),
+        index = i - 1,
+        smooth = false,
+        callback_function = function(token)
+          local guid = token.getGUID()
+          table.insert(sealedTokens, guid)
+          tokenArrangerApi.layout()
+          if name == "Bless" or name == "Curse" then
+            blessCurseManagerApi.sealedToken(name, guid)
+          end
+        end
+      })
+      return
+    end
+  end
+  printToColor(name .. " token not found in chaos bag", playerColor)
+end
+
+-- release the last sealed token
+function releaseOneToken(playerColor)
+  if not Global.call("canTouchChaosTokens") then return end
+  if sealedTokens == {} or #sealedTokens == 0 then
```

Comment:
Is sealedTokens a map or a list?  Should be one or the other and not need both checks